src/hooks/useAxiosWithAuth0.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect } from 'react';
+import axios from 'axios';
+import { useAuth0 } from '@auth0/auth0-react';
+
+/*
+NOTE:
+React requires that we can only call hooks in function components or within other hooks. As axiosWithAuth is simply a helper function we've built a useAxiosWithAuth0() hook in which we've called both axiosWithAuth AND useAuth0().
+axiosWithAuth serves the same function as it typically does. useAuth0() gives access to getAccessTokenSilently() for authentication/authorization purposes. 
+This hook below is exported and used throughout the app in place of axiosWithAuth.
+THIS useAxiosWithAuth0 HOOK SIMPLY SERVES TO HELP CONNECT THE AXIOS CALL AND THE AUTHENTICATION.
+*/
+
+export default function useAxiosWithAuth0() {
+  const { getAccessTokenSilently } = useAuth0();
+
+  const axiosWithAuth = token =>
+    axios.create({
+      baseURL: process.env.REACT_APP_API_URI,
+      headers: {
```

Comment:
Does this remove the token from local storage and attach it to the header object instead? If so, this seems like a more secure method.

Reply:
What's nice about this is actually that that token never needs to live in `local storage` to begin with! 

Here, we're pulling the token from Auth0 post-authentication by using Auth0's `getAccessTokenSilently` method and sticking it into each `axios` call's headers. 

That said, I believe that Auth0 _does_ still apply some basic token/user info to `local storage` anyway by default.